source/Halibut/Transport/Protocol/MessageExchangeProtocol.cs

```diff
@@ -40,10 +43,19 @@ public ResponseMessage ExchangeAsClient(RequestMessage request)
 
         public async Task<ResponseMessage> ExchangeAsClientAsync(RequestMessage request, CancellationToken cancellationToken)
         {
-            await PrepareExchangeAsClientAsync(cancellationToken);
+            rcpObserver.StartCall(request);
 
-            await stream.SendAsync(request, cancellationToken);
-            return await stream.ReceiveAsync<ResponseMessage>(cancellationToken);
+            try
+            {
+                await PrepareExchangeAsClientAsync(cancellationToken);
```

Comment:
I almost considered using `SendAndReceiveRequest` below here. But didn't because I thought including `PrepareExchangeAsClientAsync` as part of the 'call' made sense.